hipcheck/src/metric/entropy.rs

```diff
@@ -158,52 +158,62 @@ fn grapheme_freqs(commit_diff: &CommitDiff, db: &dyn MetricProvider) -> Result<C
 	let grapheme_table: DashMap<String, u64> = DashMap::new();
 
 	// Use this variable to track the total number of graphemes accross all patches in this commit diff.
-	let mut total_graphemes: usize = 0;
-
-	// Iterate over the file diffs by reference.
-	for file_diff in &commit_diff.diff.file_diffs {
-		// Filter out any that are probably not source files.
-		if db
-			.is_likely_source_file(Arc::clone(&file_diff.file_name))?
-			.not()
-		{
-			continue;
-		}
-
-		// Filter out any that are empty.
-		if file_diff.patch.is_empty() {
-			continue;
-		}
+	let mut res: Vec<usize> = vec![];
+	let tgt_diffs: Result<Vec<&FileDiff>> = commit_diff
+		.diff
+		.file_diffs
+		.iter()
+		.filter_map(|file_diff| {
+			// Filter out any that are probably not source files, or are empty patches
+			let is_source: bool = match db.is_likely_source_file(Arc::clone(&file_diff.file_name)) {
+				Err(e) => {
+					return Some(Err(e));
+				}
+				Ok(s) => s,
+			};
+			if !is_source || file_diff.patch.is_empty() {
+				None
+			} else {
+				Some(Ok(file_diff))
+			}
+		})
+		.collect();
+	tgt_diffs?
+		.par_iter()
+		.map(|file_diff| {
+			// Count the number of graphemes in this patch, add it to the total,
+			// and track the number of each grapheme.
+			file_diff
+				.patch
+				// Iterate in parallel over the lines of the patch.
+				.par_lines()
```

Comment:
We should probably `flat_map` to `par_lines` here, if we're trying to do everything in parallel.